src/metabase/models/params/chain_filter.clj

```diff
@@ -643,13 +656,14 @@
 
 (defn- cached-field-values-search
   [field-id query constraints {:keys [limit]}]
-  (let [values (cached-field-values field-id constraints nil)
-        query  (str/lower-case query)]
-    (cond->> (filter (fn [s]
-                       (when s
-                         (str/includes? (str/lower-case s) query)))
-                     values)
-      limit (take limit))))
+  (let [{:keys [values has_more_values]} (cached-field-values field-id constraints nil)
+        query                            (str/lower-case query)]
+    {:values (cond->> (filter (fn [s]
+                                (when s
```

Comment:
This `filter` might be clearer as `(for [... :when ...] ...)`?

Reply:
it does, but I feel like using `filter` has better sematic for `filtering` than using `for :when`.

so I'd keep using `filter` for this.